compiler/rustc_next_trait_solver/src/solve/assembly/mod.rs

```diff
@@ -288,6 +289,25 @@ where
         let Ok(normalized_self_ty) =
             self.structurally_normalize_ty(goal.param_env, goal.predicate.self_ty())
         else {
+            // HACK: Register a fake candidate when normalization fails so that
```

Comment:
```suggestion
            // FIXME: We register a fake candidate when normalization fails so that
```
I think it's fine to merge as is, but would like to look into alternative ways to do this at some point